Client/Frontend/Home/FxHomeJumpBackInViewModel.swift

```diff
@@ -24,29 +23,60 @@ struct JumpList {
 class FirefoxHomeJumpBackInViewModel: FeatureFlagsProtocol {
 
     // MARK: - Properties
+
     var jumpList = JumpList(group: nil, tabs: [Tab]())
+    var onTapGroup: ((Tab) -> Void)?
 
-    var layoutVariables: JumpBackInLayoutVariables
-    private var tabManager: TabManager
-    private var profile: Profile
-    private var isZeroSearch: Bool
-    lazy var siteImageHelper = SiteImageHelper(profile: profile)
+    private lazy var siteImageHelper = SiteImageHelper(profile: profile)
+    private let isZeroSearch: Bool
+    private let profile: Profile
+    private let tabManager: TabManager
 
-    var onTapGroup: ((Tab) -> Void)?
+    init(isZeroSearch: Bool = false,
+         profile: Profile,
+         tabManager: TabManager = BrowserViewController.foregroundBVC().tabManager) {
 
-    init(isZeroSearch: Bool) {
-        self.tabManager = BrowserViewController.foregroundBVC().tabManager
-        self.profile = BrowserViewController.foregroundBVC().profile
-        self.layoutVariables = JumpBackInLayoutVariables(columns: 1, scrollDirection: .vertical, maxItemsToDisplay: 2)
+        self.profile = profile
         self.isZeroSearch = isZeroSearch
+        self.tabManager = tabManager
     }
 
-    public func updateDataAnd(_ layoutVariables: JumpBackInLayoutVariables) {
-        self.layoutVariables = layoutVariables
-        updateJumpListData()
+    var layoutVariables: JumpBackInLayoutVariables {
+        let horizontalVariables = JumpBackInLayoutVariables(columns: 2, scrollDirection: .horizontal, maxItemsToDisplay: 4)
+        let verticalVariables = JumpBackInLayoutVariables(columns: 1, scrollDirection: .vertical, maxItemsToDisplay: 2)
+
+        let deviceIsiPad = UIDevice.current.userInterfaceIdiom == .pad
+        let deviceIsInLandscapeMode = UIWindow.isLandscape
+        let horizontalSizeClassIsCompact = UIScreen.main.traitCollection.horizontalSizeClass == .compact
+
+        if deviceIsiPad {
+            if horizontalSizeClassIsCompact { return verticalVariables }
+            return horizontalVariables
+
+        } else {
+            if deviceIsInLandscapeMode { return horizontalVariables }
+            return verticalVariables
+        }
```

Comment:
Let's not mix variables & properties and inits and functions around. I'd prefer to keep file structure a bit more organized, so it's easier to find things when we go looking for them.

Reply:
I moved this code around, but will retouch it in my next PR (regarding the layout of the jump back in section for a11y). I suggest that we discuss it there.